packages/core/src/specHelper.ts

```diff
@@ -141,11 +141,74 @@ export class PerformanceObserverStubBuilder {
   }
 }
 
+class StubXhr {
+  public response: string | undefined = undefined
+  public status: number | undefined = undefined
+  public readyState: number = XMLHttpRequest.UNSENT
+  public onreadystatechange: () => void = noop
+
+  private element: HTMLDivElement
+
+  constructor() {
+    this.element = document.createElement('div')
```

Comment:
This looks a bit weird, what about using a small event emiter implementation?

Reply:
does https://github.com/DataDog/browser-sdk/pull/565/commits/9d11fec5ac398ec0164298d61b49ef697f2e4db4 clarify it a bit?
the idea was to delegate xhr event capabilities to an other object with such features to avoid redoing it ourself.
If you see any issue with that, I can switch implementation.